pkg/builder/builder.go

```diff
@@ -476,13 +480,62 @@ func makeReadinessProbe(vdb *vapi.VerticaDB) *corev1.Probe {
 	probe := &corev1.Probe{
 		ProbeHandler: corev1.ProbeHandler{
 			Exec: &corev1.ExecAction{
-				Command: []string{"bash", "-c", buildReadinessProbeSQL(vdb)},
+				Command: []string{"bash", "-c", buildCanaryQuerySQL(vdb)},
 			},
 		},
 	}
-	ov := vdb.Spec.ReadinessProbeOverride
+	overrideProbe(probe, vdb.Spec.ReadinessProbeOverride)
+	return probe
+}
+
+// makeStartupProbe will return the Probe object to use for the startup probe.
+func makeStartupProbe(vdb *vapi.VerticaDB) *corev1.Probe {
+	probe := &corev1.Probe{
+		ProbeHandler: corev1.ProbeHandler{
+			Exec: &corev1.ExecAction{
+				Command: []string{"bash", "-c", buildCanaryQuerySQL(vdb)},
+			},
+		},
+		// We want to wait about 20 minutes for the server to come up before the
+		// other probes come into affect. The total length of the probe is more or
+		// less: InitialDelaySeconds + PeriodSeconds * FailureThreshold.
+		InitialDelaySeconds: 30,
+		PeriodSeconds:       10,
+		FailureThreshold:    117,
+		TimeoutSeconds:      5,
+	}
+	overrideProbe(probe, vdb.Spec.StartupProbeOverride)
+	return probe
+}
+
+// makeLivenessProbe will return the Probe object to use for the liveness probe.
+func makeLivenessProbe(vdb *vapi.VerticaDB) *corev1.Probe {
+	probe := &corev1.Probe{
+		// We check if the TCP client port is open. We used this approach,
+		// rather than issuing 'select 1' like readinessProbe because we need
+		// to minimize variability. If the livenessProbe fails, the pod is
+		// rescheduled. So, it isn't as forgiving as the readinessProbe.
+		ProbeHandler: corev1.ProbeHandler{
+			TCPSocket: &corev1.TCPSocketAction{
+				Port: intstr.FromInt(VerticaClientPort),
+			},
+		},
+		// These values were picked so that the vertica process needs to be
+		// unresponsive for about 1.5-minutes before it gets killed. Formula is:
+		// InitialDelaySeconds + PeriodSeconds * FailureThreshold.
```

Comment:
Based on this formula, shouldn't it be 2.5 mins? 60 + 30 * 3 = 150/60 = 2.5

Reply:
Right. I need to update the comment. It's 2.5 minutes at the start of the container, then 1.5 subsequently.